### DISCUSSION TITLE
Creating buildings for Hubs using Spoke or Blender?

### DISCUSSION THREAD

**Discussion by emclaren** [mozilla/hubs#2721] [General Questions & Discussion] [answered]
Is it more common to build architecture within 3D software (for example, a complex building) and then import the whole structure to Spoke, or to build the structure piecemeal within Spoke? For example, building parts in Blender, then importing them into Spoke to be assembled into a building.

Note: this question is copied over from Discord

**Comment by emclaren** [comment]
This is a great question, both techniques would work! However, I would likely recommend if you are building components in Blender already, that you continue using blender to assemble the full building. This would allow you to do further performance optimizations such as light baking and triangle decimation to improve performance

**Reply by hilarytsai** [reply]
Thanks, emclaren!! This is helpful. We actually are prototyping a space by building from elements in Spoke and sketchfab as well as custom assets, but we haven't really done any architecture in Blender yet.  We have several of us working in C4D too. I think we will probably end up doing a mix of building natively in Spoke and in 3D software

**Reply by davegoopot** [reply]
What's the learning curve like? (Text-based coder well out of my depth with 3D modelling here!) I've had a go a Blender in the past and found it really difficult to pick up. Would I be better using Spoke?

Seeing as I'm a coder, is there any way to code up scenes in a text format (I'm thinking similar to HTML) and get them rendered into scenes usable by Hubs?

**Reply by ocularvr** [reply]
Babylon.js also empowers 3d studio max with a free plugin for glb export that works like a dream in spoke.

**Comment by emclaren** [comment, marked as answer]
Note: I also wrote an answer related to your question comparing when I would use Spoke vs Blender more generally.

**Building architecture in Spoke**

PRO
- Easy to get started
- Web based (no downloads)
- Lots of pre-optimized building pieces available in the architecture kit
- Easy access to Sketchfab/Google Poly assets

CON
- While the architecture kit has lots of pieces, there are a limited number of shapes/textures available.


**Building architecture in Blender**

PRO
- More customizable
- More options for being able to optimize performance (e.g. reducing the number of triangles, baking the lighting)

CON
- Requires download of software
- A steeper learning curve for getting started

Generally, if someone is brand new to 3D and just wants to host an event in a nice building I recommend that they either create something using the architecture kit in Spoke or find a pre-made building on Sketchfab. Note that Sketchfab assets found through Hubs/Spoke interface are under a maximum size threshold and should perform reasonably well, if you choose to purchase a model directly off Sketchfab to import, you may need to do some work to optimize it for the web (reduce the size of the textures & triangles). 

If someone wants to create a new space from scratch & optimization and customization is important, I recommend using Blender.
